src/sysext/sysext.c

```diff
@@ -1299,6 +1328,14 @@ static int write_work_dir_file(ImageClass image_class, const char *meta_path, co
         if (!f)
                 return log_oom();
 
+        atfd = open(f, O_CLOEXEC);
+        if (atfd < 0)
+                return log_error_errno(atfd, "Failed to open '%s': %m", f);
+
+        r = mac_selinux_fix_full(atfd, NULL, hierarchy, 0);
+        if (r < 0)
+                return log_error_errno(r, "Failed to fix SELinux label for '%s': %m", f);
```

Comment:
hmm, are you sure you want to relabel f here? (and not the file you write below)